From d527a09f6acbf8a5c6a8e8bbc95e6909d42249de Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Mon, 27 Feb 2017 22:10:57 -0800
Subject: [PATCH 2312/2848] YARN-6231. FairSchedulerTestBase helper methods
 should call scheduler.update to avoid flakiness.
 (kasha)

(cherry picked from commit f187d63816584b783fbfe238475c8f37decdb6dc)
(cherry picked from commit 58eefc71c7a08b8334297a1b0057bb6dbf0978b9)

Change-Id: I9078fe9e8180bcc245625091125c2919d9c1b413
---
 .../scheduler/fair/FairSchedulerTestBase.java      |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairSchedulerTestBase.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairSchedulerTestBase.java
index 7545862..ab8a454 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairSchedulerTestBase.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairSchedulerTestBase.java
@@ -183,6 +183,7 @@ protected ApplicationAttemptId createSchedulingRequest(
         .put(id.getApplicationId(), rmApp);
 
     scheduler.allocate(id, ask, new ArrayList<ContainerId>(), null, null);
+    scheduler.update();
     return id;
   }
   
@@ -231,6 +232,7 @@ protected void createSchedulingRequestExistingApplication(
     List<ResourceRequest> ask = new ArrayList<ResourceRequest>();
     ask.add(request);
     scheduler.allocate(attId, ask, new ArrayList<ContainerId>(), null, null);
+    scheduler.update();
   }
 
   protected void createApplicationWithAMResource(ApplicationAttemptId attId,
@@ -296,4 +298,4 @@ protected void addNode(int memory, int vcores) {
     scheduler.handle(new NodeAddedSchedulerEvent(node));
     rmNodes.add(node);
   }
-}
\ No newline at end of file
+}
-- 
1.7.9.5

